walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,74 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
+/**
+ * Given a list of events, returns a list of possible time ranges for the meeting request.
+ */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Meeting duration cannot be longer than a day.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+    if (events.isEmpty() || request.getAttendees().isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    
+    List<TimeRange> eventTimes = new ArrayList<>();
+    // Collect time ranges while filtering out non-attendees.
+    for (Event event : events) {
+      if (!Collections.disjoint(event.getAttendees(), request.getAttendees())) {
+        // The event and meeting request have at least one attendee in common.
+        eventTimes.add(event.getWhen());
+      }
+    }
+    if (eventTimes.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    Collections.sort(eventTimes, TimeRange.ORDER_BY_START);
+
+    List<TimeRange> possibleTimes = new ArrayList<>();
+
+    // Check duration between start of day and start of first event.
+    if ((eventTimes.get(0).start() - TimeRange.START_OF_DAY) >= request.getDuration()) {
+      possibleTimes.add(TimeRange.fromStartEnd(
+        TimeRange.START_OF_DAY, eventTimes.get(0).start(), false)
```

Comment:
here and in line 64, could you add information about the parameter 

"If a boolean parameter can't be avoided, consider documenting the call sites using argument comments with the following format:"

`booleanMethod(/* enableFoo= */ true);`

Reply:
I edited the structure of the function in the next PR so I'll put this edit in that PR.

Reply:
^ edit: b/c of how I refactored the code I don't think this should be a problem anymore but comment on the other PR if I should still add that comment

Reply:
ok sg. i'll take a look at the next PR thanks.